api/src/main/java/org/openmrs/api/db/hibernate/HibernateCohortDAO.java

```diff
@@ -61,27 +61,36 @@ public Cohort getCohort(Integer id) throws DAOException {
 	 */
 	@Override
 	@SuppressWarnings("unchecked")
-	public List<Cohort> getCohortsContainingPatientId(Integer patientId, Boolean voided) throws DAOException {
-		Disjunction orDisjunction = Restrictions.disjunction();
-		orDisjunction.add(Restrictions.isNull("m.endDate"));
-		orDisjunction.add(Restrictions.gt("m.endDate", new Date()));
+	public List<Cohort> getCohortsContainingPatientId(Integer patientId, Boolean includeVoided,
+	                                                  Date asOfDate) throws DAOException {
+		Disjunction orStartDate = Restrictions.disjunction();
+		Disjunction orEndDate = Restrictions.disjunction();
+		orStartDate.add(Restrictions.le("m.startDate", asOfDate));
+		orStartDate.add(Restrictions.eq("m.startDate", asOfDate));
```

Comment:
You don't need this second restriction because Restrictions.le is already inclusive of the specified date